src/tools/bit2core.c

```diff
@@ -10,37 +10,42 @@
 
 const int CORE_HEADER_SIZE = 4096;
 const int BITSTREAM_HEADER_SIZE = 120;
+const int BITSTREAM_HEADER_TARGETNAME_LOC = 0x4C;
 
 unsigned char bitstream[MAX_MB*BYTES_IN_MEGABYTE];
 
 typedef struct {
   char name[8];
   int num;
   int max_core_mb_size;
+  char target_device_name[13];
 } rev_info;
 
+static rev_info revs[] =
+{
+  // name   num MB  Target Device Name
+  { "nexys", 1, 4, "7a100tcsg324" },
+  { "r2",    2, 4, "7a100tfgg484" },
+  { "r3",    3, 8, "7a200tfbg484" }
```

Comment:
I'm considering adding a "devkit" synonym here also, maybe it might help devkit owners that wouldn't know if their device was r2 or r3 ¯\_(ツ)_/¯

Reply:
I'll skip this 'mega65r3|devkit' synonym idea for now, as it'll complicate things for me, since I now assume that multiple m65target names for the same FPGA part should give the user a warning that we can't distinguish between them based on fpga part.